internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -665,6 +672,7 @@ func resourceKubernetesClusterNodePoolRead(d *pluginsdk.ResourceData, meta inter
 		d.Set("enable_node_public_ip", props.EnableNodePublicIP)
 		d.Set("enable_host_encryption", props.EnableEncryptionAtHost)
 		d.Set("fips_enabled", props.EnableFIPS)
+		d.Set("ultra_ssd_enabled", props.EnableUltraSSD)
```

Comment:
what happens if I'm using an older cluster without this field (for example provision this using a release version of the provider and then upgrade to a development version) - this'll be `nil` in the API response and not a boolean? does this need to be defaulted to false?

Reply:
It will be no changes, because it'll be `nil` in the response. But I think adding a default value makes sense to me.